lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/codeactions/EditAttribute.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+* Copyright (c) 2022 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+
+package io.openliberty.tools.langserver.lemminx.codeactions;
+
+import java.util.List;
+
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+public class EditAttribute implements ICodeActionParticipant {
+
+    @Override
+    public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+            SharedSettings sharedSettings, IComponentProvider componentProvider) {
+        try {
+            String title = "Change the optional attribute from false to true.";
+            String replaceText = "optional=\"true\"";
+            codeActions.add(CodeActionFactory.replace(title, diagnostic.getRange(), replaceText, document.getTextDocument(), diagnostic));
+
+            // also build option to create file
+            new CreateFile().doCodeAction(diagnostic, range, document, codeActions, sharedSettings, componentProvider);
```

Comment:
why?

Reply:
I thought we discussed that if a file doesn't exist on path which is required, we could provide the user the options to either create the file or make the file not required?

Reply:
I guess I thought the code action to create the file would be on the location attribute, not the optional attribute. So it seemed odd that this code was here.